Crust/Mapper/MappingProtocols.swift

```diff
@@ -1,17 +1,50 @@
 import Foundation
 import JSONValueRX
 
-public struct MappingOptions: OptionSet {
-    public let rawValue: UInt
-    public init(rawValue: UInt) {
-        self.rawValue = rawValue
+public enum CollectionInsertionMethod<Container: Sequence> {
+    case append
+    case replace(delete: ((_ orphansToDelete: Container) -> Container)?)
+}
+
+public typealias CollectionUpdatePolicy<Container: Sequence> =
+    (insert: CollectionInsertionMethod<Container>, unique: Bool)
+
+public enum Spec<T: Mapping>: Keypath {
+    
+    case mapping(Keypath, T)
+    case collectionMapping(Keypath, T, CollectionUpdatePolicy<T.SequenceKind>)
+    
+    public var keyPath: String {
+        switch self {
+        case .mapping(let keyPath, _):
+            return keyPath.keyPath
+        case .collectionMapping(let keyPath, _, _):
+            return keyPath.keyPath
+        }
+    }
+    
+    public var mapping: T {
+        switch self {
+        case .mapping(_, let mapping):
+            return mapping
+        case .collectionMapping(_, let mapping, _):
+            return mapping
+        }
+    }
+    
+    public var collectionUpdatePolicy: CollectionUpdatePolicy<T.SequenceKind> {
+        switch self {
+        case .mapping(_, _):
+            return (.append, true)
+        case .collectionMapping(_, _, let method):
+            return method
+        }
     }
-    public static let None = MappingOptions(rawValue: 0)
-    public static let AllowDuplicatesInCollection = MappingOptions(rawValue: 1)
 }
 
 public protocol Mapping {
     associatedtype MappedObject
+    associatedtype SequenceKind: Sequence = [MappedObject]
```

Comment:
If the `Mapping` is mapping to a `Sequence` of objects vs a single object this defines the type of `Sequence`. Defaults to `Array`, obviously.